homeassistant/components/tplink/__init__.py

```diff
@@ -86,7 +86,13 @@ async def _async_discovery(*_: Any) -> None:
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up TPLink from a config entry."""
     try:
-        device: SmartDevice = await Discover.discover_single(entry.data[CONF_HOST])
+        hSplit = entry.data[CONF_HOST].split(":")
```

Comment:
We shouldn't store the port in the host config entry item and probably not ask for port in the same form item as host either.

If any string splitting should be done it should be done in the config flow, since for the device api `host` doesn't include the port.

Sometimes in integrations we ask for a url which can contain both host and port eg. So there may be some leeway in the form to ask for host and port combined.

Codeowners should decide how they want the form.

Reply:
Thank you for your guidance. I modified the config flow. I still want to do host splitting, since most people will not specify a port, and the extra port field in UI will be confusing for them. I discussed that with Codeowners in python-kasa when I created PR to support passing a custom port.